src/openforms/js/components/admin/form_design/FormMetaFields.js

```diff
@@ -78,6 +79,16 @@ const FormMetaFields = ({
                     <TextInput value={name} onChange={onChange} onBlur={setFormSlug} />
                 </Field>
             </FormRow>
+            <FormRow>
+                <Field
+                    name="form.internalName"
+                    label={<FormattedMessage defaultMessage="Internal name" description="Form name field label" />}
+                    helpText={<FormattedMessage defaultMessage="Internal name/title of the form" description="Form name field help text" />}
+                    errors={errors.name}
+                >
+                    <TextInput value={internalName} onChange={onChange} onBlur={setFormSlug} />
```

Comment:
```suggestion
                    <TextInput value={internalName} onChange={onChange} />
```
Slug should be generated from public name, as it's used in URLs